### PR TITLE
[#FLW-4461, #FLW-4462, #FLW-4464] Assets management

### PR DESCRIPTION
**Pull request by DRadmir**

### CODE REVIEW

File: fearless/Modules/NewWallet/ChainAssetList/ChainAssetListInteractor.swift [hunk 1]
```diff
@@ -41,59 +44,104 @@ final class ChainAssetListInteractor {
...
+            let settings = try await chainSettingsRepository.fetchAll()
+            output?.didReceive(chainSettings: settings)
+        }
     }
 }
 
 // MARK: - ChainAssetListInteractorInput
 
 extension ChainAssetListInteractor: ChainAssetListInteractorInput {
+    var shouldRunManageAssetAnimate: Bool {
```
Comment (bnsports) [inline fearless/Modules/NewWallet/ChainAssetList/ChainAssetListInteractor.swift:128]: not right place for this code. should be inside some *Settings

File: fearless/Modules/NetworkManagment/NetworkManagmentInteractor.swift [hunk 2]
```diff
@@ -106,6 +93,13 @@ extension NetworkManagmentInteractor: NetworkManagmentInteractorInput {
         save(updatedWallet)
     }
 
+    func didSelect(_ identifire: String) {
```
Comment (bnsports) [inline fearless/Modules/NetworkManagment/NetworkManagmentInteractor.swift:96]: identifier

File: fearless/Modules/NetworkIssuesNotification/NetworkIssuesNotificationPresenter.swift [hunk 3]
```diff
@@ -146,7 +146,36 @@ extension NetworkIssuesNotificationPresenter: NetworkIssuesNotificationViewOutpu
...
 extension NetworkIssuesNotificationPresenter: NetworkIssuesNotificationInteractorOutput {
     func didReceiveChainsIssues(issues: [ChainIssue]) {
-        self.issues = issues
+        self.issues = issues.compactMap { issue in
+            guard self.issues.isNotEmpty else {
+                return issue
+            }
+            switch issue {
+            case let .network(chains):
+                let filtred = chains.filter { chain in
```
Comment (bnsports) [inline fearless/Modules/NetworkIssuesNotification/NetworkIssuesNotificationPresenter.swift:155]: filtered

File: fearless/Modules/NetworkIssuesNotification/NetworkIssuesNotificationPresenter.swift [hunk 4]
```diff
@@ -146,7 +146,36 @@ extension NetworkIssuesNotificationPresenter: NetworkIssuesNotificationViewOutpu
...
+                            return chains.contains(where: { $0.chainId == chain.chainId })
+                        } else {
+                            return false
+                        }
+                    })
+                }
+                return .network(chains: filtred)
+
+            case let .missingAccount(chains):
+                let filtred = chains.filter { chain in
```
Comment (bnsports) [inline fearless/Modules/NetworkIssuesNotification/NetworkIssuesNotificationPresenter.swift:167]: filtered

**Review by bnsports — CHANGES_REQUESTED** [review]

File: fearless/ApplicationLayer/Services/Balance/RemoteSubscription/AccountInfoRemoteService.swift [hunk 5]
```diff
@@ -0,0 +1,230 @@
...
+        case .asset:
+            let assetAccountInfo = try json.map(to: AssetAccount.self)
+            accountInfo = AccountInfo(assetAccount: assetAccountInfo)
+        }
+
+        return accountInfo
+    }
+
+    private func createSubstrateRequest(for chainAsset: ChainAsset, accountId: AccountId) -> any MixStorageRequest {
+        if chainAsset.chain.knownChainEquivalent == .genshiro {
```
[the remark this answers is not in the document]
Comment (DRadmir) [reply fearless/ApplicationLayer/Services/Balance/RemoteSubscription/AccountInfoRemoteService.swift:142]: exactly

File: fearless/Common/Helpers/EthereumNodeFetching.swift [hunk 6]
```diff
@@ -113,7 +113,9 @@ final class EthereumNodeFetching {
 
         let finalURL = ethereumChain.apiKeyInjectedURL(baseURL: wssURL)
 
-        return try Web3(wsUrl: finalURL.absoluteString).eth
+        let provider = try Web3WebSocketProvider(wsUrl: finalURL.absoluteString, timeout: .seconds(10))
+        let web3 = Web3(provider: provider, rpcId: 1)
```
[the remark this answers is not in the document]
Comment (DRadmir) [reply fearless/Common/Helpers/EthereumNodeFetching.swift:117]: correct

File: fearless/Modules/AssetManagement/AssetManagementPresenter.swift [hunk 7]
```diff
@@ -0,0 +1,247 @@
...
+                pendingAccountInfoChainAssets: pendingAccountInfoChainAssets
+            )
+            await view?.didReceive(viewModel: viewModel)
+        }
+    }
+
+    private func getInitialData() {
+        Task {
+            do {
+                let chainAssets = try await interactor.getAvailableChainAssets()
```
[the remark this answers is not in the document]
Comment (DRadmir) [reply fearless/Modules/AssetManagement/AssetManagementPresenter.swift:92]: need for next line

**Review by bnsports — APPROVED** [review]

**Pull request merged**